opm/autodiff/SimulatorFullyImplicitBlackoilOutput.cpp

```diff
@@ -296,8 +296,21 @@ namespace Opm
             }
             else {
                 // just write the data to disk
-                writeTimeStepSerial( timer, state, wellState, cellData, substep );
+                int err = 0;
+                try {
+                    writeTimeStepSerial( timer, state, wellState, cellData, substep );
+                    MPI_Bcast(&err, 1, MPI_INT, 0, MPI_COMM_WORLD);
+                } catch (std::runtime_error& msg) {
```

Comment:
This is actually a good one. First of all, MP calls should be guarded with #if HAVE_MPI and secondly, starting MPI_Bcast inside an if loop that is only entered on the master rank will most likely result in a deadlock. Did you test this?